File: saarchtestwithdifferetsimplergame/game.py
import random

class game():
    def __init__(self, pos=[]):
        if not len(pos) == 3:
            self.scramble()
        else:
            self.pos = pos

    def __str__(self):
        return str(self.pos)

    def mov1(self):
        if not self.pos[0] == 9:
            self.pos[0] += 1
        else:
            self.pos[0] = 0

    def mov2(self):
        if not self.pos[0] == 0:
            self.pos[0] -= 1
        else:
            self.pos[0] = 9
        
        if not self.pos[1] == 9:
            self.pos[1] += 1
        else:
            self.pos[1] = 0
        
        if not self.pos[2] == 0:
            self.pos[2] -= 1
        else:
            self.pos[2] = 9

    def mov3(self):
        if not self.pos[2] == 9:
            self.pos[2] += 1
        else:
            self.pos[2] = 0

    def move(self, x):
        
        if x == 0:
            self.mov1()

        elif x == 1:
            self.mov2()
        
        elif x == 2:
            self.mov3()

    def scramble(self):
        self.pos = [random.randint(0, 9), random.randint(0, 9), random.randint(0, 9)]

    def solved(self):
        return (self.pos[0] == self.pos[1]) and (self.pos[0] == self.pos[2])

def player_solve():
    g = game([3, 2, 1])

    while not g.solved():
        print(g)
        x = int(input("move: "))
        g.move(x)

    print(g)


#player_solve()
#[1, 0, 1, 2, 2], [1, 0, 1, 0, 2], [0, 0, 0, 0, 0, 0, 1, 0, 2, 2] [0, 0, 0, 0, 0, 0, 1, 0, 2, 2]

File: saarchtestwithdifferetsimplergame/cpp/game.cpp
#include <iostream>
using namespace std;

class game{
	public:
		int pos[3];
		void set(int x[3]){
			pos[0] = x[0];
			pos[1] = x[1];
			pos[2] = x[2];
		}
		
		int * get(){
			return pos;
		}
		
		void mov1(){
			if(pos[0] != 9){
				pos[0]++;
			}
			else{
				pos[0] = 0;
			}
		}
		
		void mov2(){
			if(pos[0] != 0){
				pos[0]--;
			}
			else{
				pos[0] = 9;
			}
			if(pos[1] != 9){
				pos[1]++;
			}
			else{
				pos[1] = 0;
			}
			if(pos[2] != 0){
				pos[2]--;
			}
			else{
				pos[2] = 9;
			}
		}
		
			
		void mov3(){
			if(pos[2] != 9){
				pos[2]++;
			}
			else{
				pos[2] = 0;
			}
		}
		
		void move(int m){
			if(m == 0){mov1();}
			if(m == 1){mov2();}
			if(m == 2){mov3();}
		}
		
		bool solved(){
			return (pos[0] == pos[1] && pos[0] == pos[2]);
			
		}
					
};

void print(int p[3]){
	cout << p[0] << " " << p[1] << " " << p[2] << endl;
}

int main(){
	game g;
	bool game_over = true;
	int i[3] = {4, 2, 1};
	int p[3], m;
	g.set(i);
	print(p);
	
	while (!g.solved()){
		cout << "move: ";
		cin >> m;
		
		g.move(m);
		
		p[0] = g.get()[0];
		p[1] = g.get()[1];
		p[2] = g.get()[2];
	
		print(p);
	}
}